packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts

```diff
@@ -400,6 +400,7 @@ export const useFieldV6TextField: UseFieldTextField<false> = (params) => {
       isRTL,
     );
   }, [
+    params.forwardedProps,
```

Comment:
This could force the placeholder to be recalculated more often than we'd want. 🙈